maker/src/cfd.rs

```diff
@@ -529,31 +529,38 @@ impl<O, T, W> Actor<O, T, W> {
 
         let order_id = msg.order_id;
 
-        let cfd = self.db.load_open_cfd::<Cfd>(order_id, ()).await?;
-
-        if can_use_libp2p(&cfd) {
-            // Using send here is fine because we dispatch to a task internally
-            match self
-                .libp2p_rollover
-                .send(daemon::rollover::maker::Accept {
-                    order_id,
-                    tx_fee_rate: current_offers.tx_fee_rate,
-                    long_funding_rate: current_offers.funding_rate_long,
-                    short_funding_rate: current_offers.funding_rate_short,
-                })
-                .await
-            {
-                // Return early if dispatch to libp2p rollover worked
-                Ok(Ok(())) => return Ok(()),
-                Ok(Err(error)) => {
-                    tracing::warn!("Try fallback to legacy rollover because unable to dispatch accept to libp2p rollover actor: {error:#}");
-                }
-                Err(error) => {
-                    tracing::warn!("Try fallback to legacy rollover because unable to handle accept via libp2p: {error:#}");
+        // We try to dispatch to libp2p rollover first
+        // Using send here is fine because we dispatch to a task internally
+        match self
+            .libp2p_rollover
+            .send(daemon::rollover::maker::Accept {
+                order_id,
+                tx_fee_rate: current_offers.tx_fee_rate,
+                long_funding_rate: current_offers.funding_rate_long,
+                short_funding_rate: current_offers.funding_rate_short,
+            })
+            .await
+        {
+            // Return early if dispatch to libp2p rollover worked
+            Ok(Ok(())) => return Ok(()),
+            Ok(Err(error)) => {
+                let cfd = self.db.load_open_cfd::<Cfd>(order_id, ()).await?;
+
+                if can_use_libp2p(&cfd) {
+                    tracing::error!("Failed to accept rollover for libp2p: {error:#}");
+                    return Err(error);
                 }
+
+                tracing::debug!("Try fallback to legacy rollover because unable to handle accept via libp2p: {error:#}");
+            }
+            Err(error) => {
+                // we should never see this given that the libp2p rollover actor is always running
+                tracing::error!("Try fallback to legacy rollover because unable to dispatch accept to libp2p rollover actor: {error:#}");
```

Comment:
I would just return an error here. We should not be trying to use a different protocol because maker has errors; this will just incur errors on the taker because of maker's misbehavior.

Reply:
Given that it might be a legacy scenario we have to fallback here.
We cannot just not rollover legacy requests because libp2p is broken...

But with the additional check for `peer_id` this should be fine.